apps/api/src/domain/models/photograph/repositories/arango-photograph-query-repository.spec.ts

```diff
@@ -0,0 +1,398 @@
+
+            for (const photograph of photographViews) {
+                await testQueryRepository.create(photograph);
+            }
+        });
+
+        it(`should return the expected photograph views`, async () => {
+            const result = await testQueryRepository.fetchMany();
+
+            expect(result).toHaveLength(photographViews.length);
+        });
+    });
+
+    describe(`count`, () => {
+        describe(`when there are photograph views in the database`, () => {
+            beforeEach(async () => {
+                await arangoDatabaseForCollection.clear();
+
+                for (const photograph of photographViews) {
+                    await testQueryRepository.create(photograph);
+                }
+            });
+
+            it(`should return the expected result`, async () => {
+                const result = await testQueryRepository.count();
+
+                expect(result).toBe(photographViews.length);
+            });
+        });
+
+        describe(`when the database collection is empty`, () => {
+            beforeEach(async () => {
+                await arangoDatabaseForCollection.clear();
+
+                // no photographs are added here
+            });
+
+            it(`should return 0`, async () => {
+                const result = await testQueryRepository.count();
+
+                expect(result).toBe(0);
+            });
+        });
+    });
+
+    describe(`allowUser`, () => {
+        const targetPhotograph = photographViews[0];
+
+        beforeEach(async () => {
+            // clear existing term views
+            await arangoDatabaseForCollection.clear();
+
+            await testQueryRepository.create(targetPhotograph);
+        });
+
+        it(`should add the user to the ACL`, async () => {
+            const userId = buildDummyUuid(457);
+
+            await testQueryRepository.allowUser(targetPhotograph.id, userId);
+
+            const updatedView = (await testQueryRepository.fetchById(
+                targetPhotograph.id
+            )) as PhotographViewModel;
+
+            const updatedAcl = new AccessControlList(updatedView.accessControlList);
+
+            const canUser = updatedAcl.canUser(userId);
+
+            expect(canUser).toBe(true);
+        });
+    });
+
+    describe(`delete`, () => {
+        beforeEach(async () => {
+            await arangoDatabaseForCollection.clear();
+
+            await testQueryRepository.createMany(photographViews);
+        });
+
+        it(`should remove the given term`, async () => {
+            const targetTermViewId = photographIds[0];
+
+            const expectedNumberOfTermsAfterDelete = photographViews.length - 1;
+
+            await testQueryRepository.delete(targetTermViewId);
+
+            const actualNumberOfTerms = await testQueryRepository.count();
+
+            expect(actualNumberOfTerms).toBe(expectedNumberOfTermsAfterDelete);
+        });
+    });
+
+    describe(`count`, () => {
```

Comment:
Does this one appear twice?

Reply:
Yes, good call.  It's also twice in `arango-term-query-repository.spec.ts`

Reply:
removed it from photograph and term tests.